subschemas/substrate-chain/src/resolvers/Query/parachains.ts

```diff
@@ -62,28 +68,49 @@ const extractParachainData = async (
   api: Context['api'],
   id: string,
   parachain: LinkOption | undefined,
-  lastEvents: Result,
+  lastEvents: LastEvents,
+  validators: ValidatorsInfo,
 ): Promise<Parachain> => {
-  const [leases, leasePeriod] = await Promise.all([api.query.slots?.leases?.(id) as any, getLeasePeriod(api)]);
+  const [leases, leasePeriod, optLifecycle, optPending] = await Promise.all([
+    api.query.slots?.leases?.(id) as any,
+    getLeasePeriod(api),
+    api.query.paras?.paraLifecycles?.<Option<ParaLifecycle>>(id),
+    (api.query.parasInclusion || api.query.paraInclusion || api.query.inclusion)?.pendingAvailability?.<
+      Option<CandidatePendingAvailability>
+    >(id),
+  ]);
 
   const filteredLeases = leases
     .map((opt: {isSome: any}, index: any) => (opt.isSome ? index : -1))
     .filter((period: number) => period !== -1);
   const period =
     leasePeriod?.currentLease && leases && getLeasePeriodString(bnToBn(leasePeriod.currentLease), filteredLeases);
+  const validatorInfo = getValidatorInfo(id.toString(), validators);
+  const nonVoters = getNonVoters(validators.validators, optPending?.unwrapOr(undefined));
 
   return {
-    id: id,
-    name: parachain?.text.toString(),
+    id,
+    name: parachain?.text?.toString() ?? `#${id.toString()}`,
     lease: {
-      period: period,
+      period,
       blockTime: getBlockTime(api, getBlocks(api, filteredLeases, leasePeriod)).timeStringParts,
     },
-    lifecycle: '',
+    lifecycle: optLifecycle?.unwrap().toString() ?? '',
     lastIncludedBlock: lastEvents.lastIncluded[id]?.blockNumber?.toString() ?? '',
     lastBackedBlock: lastEvents.lastBacked[id]?.blockNumber?.toString() ?? '',
     homepage: parachain?.homepage,
-    validators: undefined,
-    nonVoters: undefined,
+    validators: {
+      groupIndex: validatorInfo?.groupIndex?.toString(),
+      validators: validatorInfo?.validators?.map((v): AccountInfo => {
+        return {
+          address: v.toString(),
```

Comment:
If you notice any performance issues for adding these (validators and nonVoters) then it is ok to return `undefined` here (only for the parachains list), because we are not displaying this info in the lists. 

For the parachain resolver (a single parachain by id), we do need all the info

Reply:
For address only, no issue at all. 

For the parachain list: Including more than address requires more code and more data but I say we keep it like this (if you don't need it for now) and adapt/evolve if requirements change in the future - i.e you start to display it in the parachain list.

I'll include them on the single parachain, of course